openwisp_controller/config/base/config.py

```diff
@@ -114,6 +125,16 @@ def key(self):
         """
         return self.device.key
 
+    @cache_memoize(timeout=None, args_rewrite=get_cached_checksum_args_rewrite)
```

Comment:
I think setting a timeout, probably high, is better than not having it, since devices can be deleted but the cached checksum would stay in the cache forever.